components/automate-ui/src/app/modules/nodes/nodes-list/nodes-list.component.ts

```diff
@@ -79,7 +80,19 @@ export class NodesListComponent implements OnInit, OnDestroy {
   }
 
   filterFor(type: string, item: string): void {
-    this.nodesListFilters.push({key: type, values: [item]});
+    if (type === 'last_contact') {
+      const baseTime = item.split('T');
+      // for a last contact filter, we send in two dates - beg of day and end of day
+      // so we split on T to get the date and then just add the times ourselves
+      this.nodesListFilters.push({key: type,
+        values: [
+          baseTime[0] + 'T00:00:00Z',
+          baseTime[0] + 'T23:59:59Z'
+        ]
```

Comment:
Better to deal with objects semantically rather than raw string manipulations.
```suggestion
      this.nodesListFilters.push({key: type,
        values: [
          moment(item).startOf('day').toISOString(),
          moment(item).endOf('day').toISOString()
        ]
```